nucypher/blockchain/eth/trackers/dkg.py

```diff
@@ -127,13 +127,20 @@ def _get_start_block_number(self) -> int:
         base_block = w3.eth.get_block(base_block_number)
         average_block_time = (latest_block.timestamp - base_block.timestamp) / num_past_blocks
 
+        number_of_blocks_in_the_past = int(
+            (latest_block.timestamp - target_timestamp) / average_block_time
+        )
+
         expected_start_block = w3.eth.get_block(
-            (latest_block.timestamp - target_timestamp) // average_block_time
+            latest_block.number - number_of_blocks_in_the_past
         )
-        while expected_start_block.number > 0 and expected_start_block.timestamp > target_timestamp:
+        while (
+            expected_start_block.number > 0
+            and expected_start_block.timestamp > target_timestamp
+        ):
             expected_start_block = w3.eth.get_block(expected_start_block.number - 1)
 
-        return expected_start_block.number
+        return int(expected_start_block.number - 1)
```

Comment:
Do we need to be careful here. Could this return -1 if the expected_start_block.number is 0 - might be possible in tests?

```suggestion
        return int(expected_start_block.number - 1) if expected_start_block.number > 0 else 0
```

Reply:
Or maybe adjust the earlier logic to already accomodate the -1...

Reply:
Resolved via https://github.com/nucypher/nucypher/pull/3092/commits/958c8ff9968ccd359c2f6f427af6244127c353e8